core/SymbolRef.h

```diff
@@ -973,20 +973,32 @@ class Symbols {
         return ClassOrModuleRef::fromRaw(82);
     }
 
+    static MethodRef T_Private_Methods_DeclBuilder_abstract() {
+        return MethodRef::fromRaw(4);
+    }
+
+    static MethodRef T_Private_Methods_DeclBuilder_overridable() {
+        return MethodRef::fromRaw(5);
+    }
+
+    static MethodRef T_Private_Methods_DeclBuilder_override() {
+        return MethodRef::fromRaw(6);
+    }
+
     static ClassOrModuleRef T_Sig_WithoutRuntimeSingleton() {
         return ClassOrModuleRef::fromRaw(83);
     }
 
     static MethodRef sigWithoutRuntime() {
-        return MethodRef::fromRaw(4);
+        return MethodRef::fromRaw(7);
```

Comment:
How does those IDs work? My mental model was they are arbitrary and can go in any order and also they should not repeat

Reply:
They are not arbitrary, they must match the order that symbols are created in GlobalState::initEmpty. So if you want a method symbol to have ID 7, you must have already inserted symbols with IDs 0 through 6 in GlobalState::initEmpty already.

This also means that if in GlobalState::initEmpty you want to insert some code to enter new method symbols early in the file, before other method symbols are created later, it will have the effect of bumping all the IDs.

For this reason, some people prefer adding new symbols at the end of the list and just not caring about code organization in GlobalState::initEmpty. But I think it's usually important to group similar definitions near each other, and so I typically renumber these IDs.